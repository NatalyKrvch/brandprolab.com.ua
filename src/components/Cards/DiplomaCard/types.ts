export interface DiplomaCardProps {
  diplomaURL: string;
  onClick?: () => void;
}
