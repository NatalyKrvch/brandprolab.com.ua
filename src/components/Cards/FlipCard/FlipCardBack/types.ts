export interface FlipCardBackProps {
  header: string;
  list: string[];
}
