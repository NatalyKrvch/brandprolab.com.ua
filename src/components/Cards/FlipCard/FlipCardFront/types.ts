export interface FlipCardFrontProps {
  iconURL: string;
  header: string;
  label: string;
  description: string;
  callToAction: string;
}
