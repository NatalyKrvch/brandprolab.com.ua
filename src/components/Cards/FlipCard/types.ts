import type { ReactNode } from 'react';

export interface FlipCardProps {
  front: ReactNode;
  back: ReactNode;
}
