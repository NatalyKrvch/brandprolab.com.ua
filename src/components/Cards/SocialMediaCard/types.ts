export enum MainSocialCardColor {
  RED = 'red',
  BLUE = 'blue',
}

export type socialMediaColorClassMapType = {
  [key in MainSocialCardColor]: {
    bg: string;
    text: string;
  };
};

export interface SocialMediaCardProps {
  iconURL: string;
  socialMediaURL: string;
  label: string;
  description: string;
  mainColor: MainSocialCardColor;
}
