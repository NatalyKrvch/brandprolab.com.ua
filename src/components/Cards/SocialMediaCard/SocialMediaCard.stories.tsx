import type { Meta, StoryObj } from '@storybook/react';

import { MainSocialCardColor } from '@/styles/constants';

import SocialMediaCard from './SocialMediaCard';

const meta: Meta<typeof SocialMediaCard> = {
  title: 'Components/SocialMediaCard',
  component: SocialMediaCard,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    iconURL: {
      control: 'text',
      description: 'The URL of the icon to be displayed inside the circle.',
      table: { type: { summary: 'string' } },
    },
    socialMediaURL: {
      control: 'text',
      description: 'URL to open when card is clicked.',
      table: { type: { summary: 'string' } },
    },
    label: {
      control: 'text',
      description: 'Title or label of the social media platform.',
      table: { type: { summary: 'string' } },
    },
    description: {
      control: 'text',
      description: 'Short description or message for the card.',
      table: { type: { summary: 'string' } },
    },
    mainColor: {
      control: { type: 'select' },
      options: Object.values(MainSocialCardColor),
      description: 'Main theme color for text and icon background.',
      table: { type: { summary: 'MainSocialCardColor' } },
    },
  },
};

export default meta;
type Story = StoryObj<typeof SocialMediaCard>;

export const Default: Story = {
  args: {
    iconURL: '/icons/cards/socialMedia/insta.svg',
    socialMediaURL: 'https://www.instagram.com/',
    label: 'Instagram',
    description: 'Stay updated with the latest design & brand tips.',
    mainColor: MainSocialCardColor.BLUE,
  },
};
