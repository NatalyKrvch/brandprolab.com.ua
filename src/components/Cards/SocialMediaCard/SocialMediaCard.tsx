import clsx from 'clsx';

import { Icon } from '@/components/Icon';
import { ExternalLink } from '@/components/Links';
import { IconClassType } from '@/lib/constants';
import {
  SOCIAL_MEDIA_DESCRIPTION_TEST_ID,
  SOCIAL_MEDIA_LABEL_TEST_ID,
} from '@/lib/testIDs';
import { MainSocialCardColor } from '@/styles/constants';

import { SimpleCard } from '../SimpleCard';
import type { SocialMediaCardProps } from './types';

const SocialMediaCard = ({
  iconURL,
  socialMediaURL,
  label,
  description,
  mainColor,
}: SocialMediaCardProps) => {
  const bgColorClass =
    mainColor === MainSocialCardColor.RED
      ? 'bg-red-default'
      : mainColor === MainSocialCardColor.BLUE
        ? 'bg-blue-default'
        : '';

  const textColorClass =
    mainColor === MainSocialCardColor.RED
      ? 'text-red-default'
      : mainColor === MainSocialCardColor.BLUE
        ? 'text-blue-default'
        : '';

  return (
    <ExternalLink href={socialMediaURL}>
      <SimpleCard
        backgroundColor="bg-gray-light"
        className="flex h-full flex-row items-center gap-5 px-4 pb-6 pt-4 tablet:flex-col tablet:items-start desktop:flex-col desktop:items-start"
      >
        <Icon
          iconURL={iconURL}
          circleColor={bgColorClass}
          type={IconClassType.SOCIAL_MEDIA}
        />

        <div>
          <h3
            className={clsx(
              'mb-1 text-base font-medium leading-5 tracking-wide tablet:text-base desktop:text-lg desktop:leading-22',
              textColorClass,
            )}
            data-testid={SOCIAL_MEDIA_LABEL_TEST_ID}
          >
            {label}
          </h3>

          <p
            className="text-sm font-normal leading-4 tracking-wide text-gray-dark desktop:text-base desktop:leading-18"
            data-testid={SOCIAL_MEDIA_DESCRIPTION_TEST_ID}
          >
            {description}
          </p>
        </div>
      </SimpleCard>
    </ExternalLink>
  );
};

export default SocialMediaCard;
