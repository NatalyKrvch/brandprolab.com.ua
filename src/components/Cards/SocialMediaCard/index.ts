export { default as SocialMediaCard } from './SocialMediaCard';
export * from './types';
