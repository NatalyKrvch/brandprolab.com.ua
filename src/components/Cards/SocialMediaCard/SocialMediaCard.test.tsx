import { render, screen } from '@testing-library/react';

import {
  EXTERNAL_LINK_TEST_ID,
  SOCIAL_MEDIA_DESCRIPTION_TEST_ID,
  SOCIAL_MEDIA_LABEL_TEST_ID,
} from '@/lib/testIDs';

import SocialMediaCard from './SocialMediaCard';
import { MainSocialCardColor } from './types';

describe('SocialMediaCard', () => {
  const defaultProps = {
    iconURL: '/icons/youtube.svg',
    socialMediaURL: 'https://youtube.com',
    label: 'Viktoriia Zakharova',
    description: 'Відео про саморозвиток і пошук роботи.',
    mainColor: MainSocialCardColor.RED,
  };

  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('renders correctly with given props', () => {
    const { container } = render(<SocialMediaCard {...defaultProps} />);

    expect(screen.getByTestId(EXTERNAL_LINK_TEST_ID)).toBeInTheDocument();
    expect(screen.getByTestId(SOCIAL_MEDIA_LABEL_TEST_ID)).toHaveTextContent(
      defaultProps.label,
    );
    expect(
      screen.getByTestId(SOCIAL_MEDIA_DESCRIPTION_TEST_ID),
    ).toHaveTextContent(defaultProps.description);
    expect(container).toMatchSnapshot();
  });

  it('applies correct href to link', () => {
    const { container } = render(<SocialMediaCard {...defaultProps} />);

    const link = screen.getByTestId(EXTERNAL_LINK_TEST_ID);
    expect(link).toHaveAttribute('href', defaultProps.socialMediaURL);
    expect(container).toMatchSnapshot();
  });

  it('renders correct color class based on mainColor prop', () => {
    const { container } = render(<SocialMediaCard {...defaultProps} />);

    const label = screen.getByTestId(SOCIAL_MEDIA_LABEL_TEST_ID);
    expect(label.className).toContain('text-red-default');
    expect(container).toMatchSnapshot();
  });
});
