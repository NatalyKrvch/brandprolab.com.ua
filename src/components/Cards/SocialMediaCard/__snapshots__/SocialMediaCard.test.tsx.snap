// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaCard applies correct href to link 1`] = `
<div>
  <a
    data-testid="external-link-test-id"
    href="https://youtube.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="bg-gray-light rounded-24 tablet:rounded-32 desktop:rounded-40 w-full bg-cover bg-center flex flex-row items-center gap-5 px-4 pb-6 pt-4 tablet:flex-col tablet:items-start desktop:flex-col desktop:items-start"
      data-testid="simple-card-test-id"
    >
      <div
        class="flex items-center justify-center"
        data-testid="icon-component-test-id"
      >
        <div
          class="bg-red-default relative flex items-center justify-center rounded-full w-36 h-36 tablet:w-42 tablet:h-42 desktop:w-48 desktop:h-48"
        >
          <img
            alt="Icon"
            class="absolute w-22 h-22 tablet:w-28 tablet:h-28 desktop:w-32 desktop:h-32"
            data-nimg="1"
            data-testid="icon-component-image-test-id"
            decoding="async"
            height="0"
            loading="lazy"
            src="/icons/youtube.svg"
            style="color: transparent;"
            width="0"
          />
        </div>
      </div>
      <div>
        <h3
          class="mb-1 text-base font-medium leading-5 tracking-wide tablet:text-base desktop:text-lg desktop:leading-22 text-red-default"
          data-testid="social-media-label-test-id"
        >
          Viktoriia Zakharova
        </h3>
        <p
          class="text-sm font-normal leading-4 tracking-wide text-gray-dark desktop:text-base desktop:leading-18"
          data-testid="social-media-description-test-id"
        >
          Відео про саморозвиток і пошук роботи.
        </p>
      </div>
    </div>
  </a>
</div>
`;

exports[`SocialMediaCard renders correct color class based on mainColor prop 1`] = `
<div>
  <a
    data-testid="external-link-test-id"
    href="https://youtube.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="bg-gray-light rounded-24 tablet:rounded-32 desktop:rounded-40 w-full bg-cover bg-center flex flex-row items-center gap-5 px-4 pb-6 pt-4 tablet:flex-col tablet:items-start desktop:flex-col desktop:items-start"
      data-testid="simple-card-test-id"
    >
      <div
        class="flex items-center justify-center"
        data-testid="icon-component-test-id"
      >
        <div
          class="bg-red-default relative flex items-center justify-center rounded-full w-36 h-36 tablet:w-42 tablet:h-42 desktop:w-48 desktop:h-48"
        >
          <img
            alt="Icon"
            class="absolute w-22 h-22 tablet:w-28 tablet:h-28 desktop:w-32 desktop:h-32"
            data-nimg="1"
            data-testid="icon-component-image-test-id"
            decoding="async"
            height="0"
            loading="lazy"
            src="/icons/youtube.svg"
            style="color: transparent;"
            width="0"
          />
        </div>
      </div>
      <div>
        <h3
          class="mb-1 text-base font-medium leading-5 tracking-wide tablet:text-base desktop:text-lg desktop:leading-22 text-red-default"
          data-testid="social-media-label-test-id"
        >
          Viktoriia Zakharova
        </h3>
        <p
          class="text-sm font-normal leading-4 tracking-wide text-gray-dark desktop:text-base desktop:leading-18"
          data-testid="social-media-description-test-id"
        >
          Відео про саморозвиток і пошук роботи.
        </p>
      </div>
    </div>
  </a>
</div>
`;

exports[`SocialMediaCard renders correctly with given props 1`] = `
<div>
  <a
    data-testid="external-link-test-id"
    href="https://youtube.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="bg-gray-light rounded-24 tablet:rounded-32 desktop:rounded-40 w-full bg-cover bg-center flex flex-row items-center gap-5 px-4 pb-6 pt-4 tablet:flex-col tablet:items-start desktop:flex-col desktop:items-start"
      data-testid="simple-card-test-id"
    >
      <div
        class="flex items-center justify-center"
        data-testid="icon-component-test-id"
      >
        <div
          class="bg-red-default relative flex items-center justify-center rounded-full w-36 h-36 tablet:w-42 tablet:h-42 desktop:w-48 desktop:h-48"
        >
          <img
            alt="Icon"
            class="absolute w-22 h-22 tablet:w-28 tablet:h-28 desktop:w-32 desktop:h-32"
            data-nimg="1"
            data-testid="icon-component-image-test-id"
            decoding="async"
            height="0"
            loading="lazy"
            src="/icons/youtube.svg"
            style="color: transparent;"
            width="0"
          />
        </div>
      </div>
      <div>
        <h3
          class="mb-1 text-base font-medium leading-5 tracking-wide tablet:text-base desktop:text-lg desktop:leading-22 text-red-default"
          data-testid="social-media-label-test-id"
        >
          Viktoriia Zakharova
        </h3>
        <p
          class="text-sm font-normal leading-4 tracking-wide text-gray-dark desktop:text-base desktop:leading-18"
          data-testid="social-media-description-test-id"
        >
          Відео про саморозвиток і пошук роботи.
        </p>
      </div>
    </div>
  </a>
</div>
`;
